webapp/channels/src/actions/views/drafts.ts

```diff
@@ -154,7 +154,9 @@ export function setDraftsTourTipPreference(initializationState: Record<string, b
 
 export function setGlobalDraft(key: string, value: PostDraft|null, isRemote: boolean) {
     return (dispatch: DispatchFunc) => {
-        dispatch(setGlobalItem(key, value));
+        if (value?.message === '' || value?.message.replace(/\s/g, '').length || (value?.message.replace(/\s/g, '').length && value?.fileInfos.length > 0)) {
+            dispatch(setGlobalItem(key, value));
+        }
```

Comment:
i do not understand this condition, why are we dispatching when message is empty?

Reply:
When we remove the draft, the message comes empty, if it's not for this, the draft will not get deleted as the action will not be dispatched.